spec/javascripts/character_counter_spec.js

```diff
@@ -0,0 +1,102 @@
+//= require modules/character-counter
+
+describe('Character counter', function() {
+
+  var $ = window.jQuery,
+      $textbox,
+      $counter;
+  
+  beforeEach(function () {
+    $textbox = $('<textarea data-max-length="50" id="fixture"></textarea>');
+    $counter = $('<p class="character-count">50 characters max</p>');
+    $(document.body).append($textbox);
+    $(document.body).append($counter);
+  });
+
+  afterEach(function () {
+    $textbox.remove();
+    $counter.remove();
+  });
+
+  describe("When called", function () {
+    it("Should put the max characters into the character counter", function () {
+      GOVUK.GDM.characterCounter();
+      expect($counter.text()).toEqual("50");
+    });
+
+    it("Should add an id to the character counter based on the textbox id", function () {
```

Comment:
describing specs, don't use 'should'  - http://betterspecs.org/#should (is valid also with jasmine specs)